### DISCUSSION TITLE
When we load a route , it throws 500 error in the deployment link but same route works fine locally.

### DISCUSSION THREAD

**Discussion by Savyashree-PW15003** [vercel/next.js#59233] [App Router]
Hi folks,

when i try to load a route that uses SSR in next js, it throws 500 - internal server error in the vercel deployment url but when i run the same branch locally using the same env variables, it works fine.

could you please help me resolving this issue?

**Comment by icyJoseph** [comment]
Put logs on the parts that fetch data, and monitor those in the Vercel dashboard.

Have you made sure to place the env vars in the Vercel dashboard? or somehow make them available to it?

Are you making assumptions about how the code is run? Does it work on your machine if you do `npm run build && npm run start` (or with yarn or pnpm)?

**Reply by Savyashree-PW15003** [reply]
Have you made sure to place the env vars in the Vercel dashboard? - yes
Are you making assumptions about how the code is run? Does it work on your machine if you do npm run build && npm run start (or with yarn or pnpm)? -  yes it works fine
fails in deployment url

**Reply by icyJoseph** [reply]
Then you must have some logs to investigate the issue. It could literally be anything...

One more thing regarding local dev, can you delete `.next`, and then run `npm run build && npm run start`?

**Reply by Savyashree-PW15003** [reply]
hey @icyJoseph , tried this, but it works fine in localhost

**Reply by icyJoseph** [reply]
Then we need to see logs from the deployment. Got anything to share? https://vercel.com/docs/observability/runtime-logs

**Reply by Savyashree-PW15003** [reply]
it shows this Error: {"error":"Route not found"}
    at Object.getProperError (/var/task/node_modules/next/dist/lib/is-error.js:25:12)

**Reply by icyJoseph** [reply]
Well, are you trying to redirect or something?

I think you'll have to start to put logs in your application, to try and trace what is not working correctly. 

The build step goes well right (on the Vercel environment)

Also, does the Vercel Node.js version match your local version?

**Comment by ayoubben18** [comment]
Have you solved the error? I got the same error right now and I've been struggling for hours!

**Reply by mr-zlaam** [reply]
have  you solved this issue
